engines/bops_admin/app/controllers/bops_admin/application_type_requirements_controller.rb

```diff
@@ -0,0 +1,88 @@
+    before_action :set_application_type
+
+    def edit
+      respond_to do |format|
+        format.html
+        @categories = LocalAuthority::Requirement.categories
+        @requirements = current_local_authority.requirements
+        @application_type = current_local_authority.application_types.find(application_type_id)
+        # raise
+      end
+    end
+
+    # def create
+    #   respond_to do |format|
+    #     format.html do
+    #       @application_type = current_local_authority.application_types.find(application_type_id)
+    #       @requirements = current_local_authority.requirements
+    #       @requirements.where(id: params[:requirement_ids]).find_each do |requirement|
+    #         application_type_requirement = ApplicationTypeRequirement.find_or_initialize_by(
+    #             application_type_id: @application_type.id,
+    #             local_authority_requirement_id: requirement.id
+    #           )
+    #         application_type_requirement.save!
+    #       end
+    #         redirect_to application_type_path(@application_type), notice: t(".success")
+    #     end
+    #   end
+    # end
+
+    def update
+      # tailored method to add and remove based on selection/ deselection
+      selected_requirement_ids = (params[:application_type][:requirement_ids]).reject { |id| id.blank? || id.to_i == 0 }.map(&:to_i)
+      # create array of existing :local_authority_requirement_ids
+      existing_requirement_ids = @application_type.application_type_requirements.pluck(:local_authority_requirement_id)
+
+      # add new ones
+      (selected_requirement_ids - existing_requirement_ids).each do |id|
+        ApplicationTypeRequirement.create!(
+          application_type_id: @application_type.id,
+          local_authority_requirement_id: id
+        )
+      end
+
+      # destroy removed ones
+      (existing_requirement_ids - selected_requirement_ids).each do |id|
+        ApplicationTypeRequirement.where(
+          application_type_id: @application_type.id,
+          local_authority_requirement_id: id
+        ).destroy_all
+      end
+
+      redirect_to application_type_path(@application_type), notice: t(".success")
+    end
+
+    private
+
+    def application_type_id
+      Integer(params[:application_type_id])
+    end
+
+    def set_application_type
+      @application_type = current_local_authority.application_types.find(application_type_id)
+    end
+
+    def set_local_authority_requirements
+      @requirements = current_local_authority.requirements
+    end
+
+    def set_application_type_requirements
+      @application_type_requirements = ApplicationTypeRequirement.includes(:local_authority_requirement).where(
+        local_authority_requirement: {local_authority_id: current_local_authority.id},
+        application_type_id: application_type_id
+      )
+    end
+
+    def set_application_type_requirement
+      @application_type_requirement = ApplicationTypeRequirement.where(
```

Comment:
This might need to be `find_by`, not `where`, if it needs a single result and not a collection.